applications/FreeSurfaceApplication/custom_utilities/edgebased_levelset.h

```diff
@@ -1202,2538 +1120,2442 @@ class EdgeBasedLevelSet
-
-        }
+            // fill the list of slip nodes
+            std::vector<unsigned int> tempmSlipBoundaryList;
+            for (unsigned int i_node = 0; i_node < n_nodes; i_node++)
+            {
+                if (mis_slip[i_node] == true)
+                    tempmSlipBoundaryList.push_back(i_node);
 
+                mis_slip[i_node] = false;
+            }
+            mSlipBoundaryList.resize(tempmSlipBoundaryList.size(), false);
+#pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(tempmSlipBoundaryList.size()); i++)
+                mSlipBoundaryList[i] = tempmSlipBoundaryList[i];
 
-        //calculating limitor
-        #pragma omp parallel for
-        for (int i_node = 0; i_node < n_nodes; i_node++)
-        {
+            // loop over all faces to fill inlet outlet
 
-            const array_1d<double, TDim>& pi_i = mPiConvection[i_node];
-            const double& p_i = mphi_n1[i_node];
-            double& beta_i = mBeta[i_node];
-            beta_i = 0.0;
-            double n = 0.0;
-            for (unsigned int csr_index = mr_matrix_container.GetRowStartIndex()[i_node]; csr_index != mr_matrix_container.GetRowStartIndex()[i_node + 1]; csr_index++)
+            for (ModelPart::ConditionsContainerType::iterator cond_it = rConditions.begin(); cond_it != rConditions.end(); cond_it++)
             {
-                unsigned int j_neighbour = mr_matrix_container.GetColumnIndex()[csr_index];
-
-                const double& p_j = mphi_n1[j_neighbour];
-                const array_1d<double, TDim>& l_k = mEdgeDimensions[csr_index];
-                const array_1d<double, TDim>& pi_j = mPiConvection[j_neighbour];
+                // get geometry data of the face
+                Geometry<Node<3>> &face_geometry = cond_it->GetGeometry();
 
-                //                                 double proj = 0.0;
-                //                                 for (unsigned int comp = 0; comp < TDim; comp++)
-                //                                      proj += 0.5*l_k[comp]*(pi_i[comp]+pi_j[comp]);
-                //                                 double beta = fabs((p_i - p_j - proj)/(fabs(p_i-p_j)+fabs(proj)+1e-4));
+                // reference for area normal of the face
+                array_1d<double, 3> &face_normal = cond_it->GetValue(NORMAL);
 
-                double proj = 0.0;
-                for (unsigned int comp = 0; comp < TDim; comp++)
-                    proj += 0.5 * l_k[comp]*(pi_i[comp] + pi_j[comp]);
-                // 							proj += dir[comp]*pi_i[comp];
+                // inlet or outlet condition
+                bool is_inlet_or_outlet = false;
+                if (cond_it->IsNot(SLIP))
+                    is_inlet_or_outlet = true;
+                else
+                {
+                    for (unsigned int if_node = 0; if_node < TDim; if_node++)
+                        if (face_geometry[if_node].Is(INLET) || face_geometry[if_node].Is(OUTLET))
```

Comment:
Same comment the `OUTLET` flag check vs. `PRESSURE` fixity check.

Reply:
Done